# Notebook for matches

### Importing Libraries

In [3]:
%pylab inline
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

Populating the interactive namespace from numpy and matplotlib


### Importing Data

In [4]:
matches = pd.read_csv("/Coursera_Work/Data Analysis Projects/IPL/Data/matches.csv")

In [5]:
matches.head()

id    Season       city        date                        team1  \
0   1  IPL-2017  Hyderabad  05-04-2017          Sunrisers Hyderabad   
1   2  IPL-2017       Pune  06-04-2017               Mumbai Indians   
2   3  IPL-2017     Rajkot  07-04-2017                Gujarat Lions   
3   4  IPL-2017     Indore  08-04-2017       Rising Pune Supergiant   
4   5  IPL-2017  Bangalore  08-04-2017  Royal Challengers Bangalore   

                         team2                  toss_winner toss_decision  \
0  Royal Challengers Bangalore  Royal Challengers Bangalore         field   
1       Rising Pune Supergiant       Rising Pune Supergiant         field   
2        Kolkata Knight Riders        Kolkata Knight Riders         field   
3              Kings XI Punjab              Kings XI Punjab         field   
4             Delhi Daredevils  Royal Challengers Bangalore           bat   

   result  dl_applied                       winner  win_by_runs  \
0  normal           0          Sunrisers Hyderabad           35   
1  normal           0       Rising Pune Supergiant            0   
2  normal           0        Kolkata Knight Riders            0   
3  normal           0              Kings XI Punjab            0   
4  normal           0  Royal Challengers Bangalore           15   

   win_by_wickets player_of_match                                      venue  \
0               0    Yuvraj Singh  Rajiv Gandhi International Stadium, Uppal   
1               7       SPD Smith    Maharashtra Cricket Association Stadium   
2              10         CA Lynn     Saurashtra Cricket Association Stadium   
3               6      GJ Maxwell                     Holkar Cricket Stadium   
4               0       KM Jadhav                      M Chinnaswamy Stadium   

          umpire1        umpire2 umpire3  
0     AY Dandekar       NJ Llong     NaN  
1  A Nand Kishore         S Ravi     NaN  
2     Nitin Menon      CK Nandan     NaN  
3    AK Chaudhary  C Shamshuddin     NaN  
4             NaN            NaN     NaN

In [6]:
matches.Season.unique()

array(['IPL-2017', 'IPL-2008', 'IPL-2009', 'IPL-2010', 'IPL-2011',
       'IPL-2012', 'IPL-2013', 'IPL-2014', 'IPL-2015', 'IPL-2016',
       'IPL-2018', 'IPL-2019'], dtype=object)

In [7]:
MI = matches[(matches['winner'] == "Mumbai Indians")]

In [10]:
MI.head()

id    Season       city        date                        team1  \
6    7  IPL-2017     Mumbai  09-04-2017        Kolkata Knight Riders   
9   10  IPL-2017     Mumbai  12-04-2017          Sunrisers Hyderabad   
11  12  IPL-2017  Bangalore  14-04-2017  Royal Challengers Bangalore   
15  16  IPL-2017     Mumbai  16-04-2017                Gujarat Lions   
21  22  IPL-2017     Indore  20-04-2017              Kings XI Punjab   

             team2     toss_winner toss_decision  result  dl_applied  \
6   Mumbai Indians  Mumbai Indians         field  normal           0   
9   Mumbai Indians  Mumbai Indians         field  normal           0   
11  Mumbai Indians  Mumbai Indians         field  normal           0   
15  Mumbai Indians  Mumbai Indians         field  normal           0   
21  Mumbai Indians  Mumbai Indians         field  normal           0   

            winner  win_by_runs  win_by_wickets player_of_match  \
6   Mumbai Indians            0               4          N Rana   
9   Mumbai Indians            0               4       JJ Bumrah   
11  Mumbai Indians            0               4      KA Pollard   
15  Mumbai Indians            0               6          N Rana   
21  Mumbai Indians            0               8      JC Buttler   

                     venue                umpire1        umpire2 umpire3  
6         Wankhede Stadium            Nitin Menon      CK Nandan     NaN  
9         Wankhede Stadium            Nitin Menon      CK Nandan     NaN  
11   M Chinnaswamy Stadium  KN Ananthapadmanabhan   AK Chaudhary     NaN  
15        Wankhede Stadium         A Nand Kishore         S Ravi     NaN  
21  Holkar Cricket Stadium              M Erasmus  C Shamshuddin     NaN

In [9]:
MI.count()

id                 109
Season             109
city               109
date               109
team1              109
team2              109
toss_winner        109
toss_decision      109
result             109
dl_applied         109
winner             109
win_by_runs        109
win_by_wickets     109
player_of_match    109
venue              109
umpire1            109
umpire2            109
umpire3             17
dtype: int64

# Mumbai has won 109 matches from IPL - 2008 to IPL - 2019

In [11]:
home_ground = MI[(MI['venue'] == 'Wankhede Stadium')]
home_ground.head()

id    Season    city        date                        team1  \
6    7  IPL-2017  Mumbai  09-04-2017        Kolkata Knight Riders   
9   10  IPL-2017  Mumbai  12-04-2017          Sunrisers Hyderabad   
15  16  IPL-2017  Mumbai  16-04-2017                Gujarat Lions   
23  24  IPL-2017  Mumbai  22-04-2017               Mumbai Indians   
36  37  IPL-2017  Mumbai  01-05-2017  Royal Challengers Bangalore   

               team2                  toss_winner toss_decision  result  \
6     Mumbai Indians               Mumbai Indians         field  normal   
9     Mumbai Indians               Mumbai Indians         field  normal   
15    Mumbai Indians               Mumbai Indians         field  normal   
23  Delhi Daredevils             Delhi Daredevils         field  normal   
36    Mumbai Indians  Royal Challengers Bangalore           bat  normal   

    dl_applied          winner  win_by_runs  win_by_wickets player_of_match  \
6            0  Mumbai Indians            0               4          N Rana   
9            0  Mumbai Indians            0               4       JJ Bumrah   
15           0  Mumbai Indians            0               6          N Rana   
23           0  Mumbai Indians           14               0  MJ McClenaghan   
36           0  Mumbai Indians            0               5       RG Sharma   

               venue         umpire1      umpire2 umpire3  
6   Wankhede Stadium     Nitin Menon    CK Nandan     NaN  
9   Wankhede Stadium     Nitin Menon    CK Nandan     NaN  
15  Wankhede Stadium  A Nand Kishore       S Ravi     NaN  
23  Wankhede Stadium  A Nand Kishore       S Ravi     NaN  
36  Wankhede Stadium    AK Chaudhary  CB Gaffaney     NaN

In [12]:
home_ground.count()

id                 42
Season             42
city               42
date               42
team1              42
team2              42
toss_winner        42
toss_decision      42
result             42
dl_applied         42
winner             42
win_by_runs        42
win_by_wickets     42
player_of_match    42
venue              42
umpire1            42
umpire2            42
umpire3             8
dtype: int64

## 42 wins came on Home ground

In [13]:
matches.describe()

id  dl_applied  win_by_runs  win_by_wickets
count    756.000000  756.000000   756.000000      756.000000
mean    1792.178571    0.025132    13.283069        3.350529
std     3464.478148    0.156630    23.471144        3.387963
min        1.000000    0.000000     0.000000        0.000000
25%      189.750000    0.000000     0.000000        0.000000
50%      378.500000    0.000000     0.000000        4.000000
75%      567.250000    0.000000    19.000000        6.000000
max    11415.000000    1.000000   146.000000       10.000000